### PR TITLE
fix: Compiler warning in src/tests/algorithms_test/pid_MergeParticleID.cc

### PR DESCRIPTION
**Pull request by c-dilks**
### Briefly, what does this PR introduce?
Resolve warning: non-void lambda does not return a value in all control paths.

### What kind of change does this PR introduce?
- [x] Bug fix (issue #927)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
no

### Does this PR change default behavior?
no

### CODE REVIEW

File: src/tests/algorithms_test/pid_MergeParticleID.cc [hunk 1]
```diff
@@ -125,6 +125,9 @@ TEST_CASE("the PID MergeParticleID algorithm runs", "[MergeParticleID]") {
         return obj;
     }
     FAIL("ERROR: cannot find CherenkovParticleID given track");
+    if(coll->size() == 0)
```
Comment (wdconinc) [inline src/tests/algorithms_test/pid_MergeParticleID.cc:128]: How nitpicky should I be on code that should never run?
```suggestion
    if(coll->empty())
```

**Review by wdconinc — APPROVED** [review]

**Review by veprbl — DISMISSED** [review]

File: src/tests/algorithms_test/pid_MergeParticleID.cc [hunk 2]
```diff
@@ -125,6 +125,9 @@ TEST_CASE("the PID MergeParticleID algorithm runs", "[MergeParticleID]") {
         return obj;
     }
     FAIL("ERROR: cannot find CherenkovParticleID given track");
+    if(coll->empty())
```
Comment (veprbl) [inline src/tests/algorithms_test/pid_MergeParticleID.cc:128]: Surprisingly, despite PODIO declaring that they try to mimic STL collections, there is no `empty()`: https://github.com/AIDASoft/podio/blob/master/python/templates/Collection.h.jinja2

Let's revert
```suggestion
    if(coll->size() == 0)
```

**Review by veprbl — APPROVED** [review]

File: src/tests/algorithms_test/pid_MergeParticleID.cc (hunk 2 above)
Comment (wdconinc) [reply src/tests/algorithms_test/pid_MergeParticleID.cc]: Ah shoot, yes, it's not even the first time I've suggested this in a code review and we've had to revert.

Comment (wdconinc) [reply src/tests/algorithms_test/pid_MergeParticleID.cc]: https://github.com/AIDASoft/podio/issues/476

**Comment by veprbl** [comment]
Evicted from the merge queue: https://github.com/eic/EICrecon/actions/runs/6115159554/job/16598662329

**Comment by wdconinc** [comment]
> Evicted from the merge queue

But clearly unrelated. This must be due to something else (and it must affects main too). Only other change is eic-shell got updated with the new EDM4eic data model, v3.0.1, and we now patch JANA2. My money is on the data model...

**Comment by veprbl** [comment]
> > Evicted from the merge queue
> 
> But clearly unrelated. This must be due to something else (and it must affects main too). Only other change is eic-shell got updated with the new EDM4eic data model, v3.0.1, and we now patch JANA2. My money is on the data model...

Look at frame `#5` https://github.com/eic/EICrecon/actions/runs/6115159554/job/16598662329#step:5:715
could be a coincidence, of course

**Comment by veprbl** [comment]
Though, indeed, this PR does not touch algorithm itself, and tests should not even get loaded in eicrecon-gun job.

**Pull request merged**